### PR TITLE
Add auto-compiled models to What's New

### PR DESCRIPTION
**Pull request by ajcvickers**

### CODE REVIEW

**Comment by ajcvickers** [comment]
## Release notes

The team has been working primarily on EF Core internals, so there are no new big features in EF Core 9 (EF9) Preview 3. However, this means we really need people like you to run your code on these new internals and report back what you find. We want to fix bugs in the new internals as soon as possible in order to have a strong GA release later in the year.

That being said, there are several smaller enhancements included in preview 3. For full details, see [What's new in EF Core 9](https://learn.microsoft.com/ef/core/what-is-new/ef-core-9.0/whatsnew). To discuss the EF9 release, go to [Try EF Core 9 now!](https://github.com/dotnet/efcore/issues/33030) on GitHub.

### Auto-compiled models

Auto-compiled models allow the [EF Core compiled model](https://learn.microsoft.com/ef/core/performance/advanced-performance-topics#compiled-models) to be automatically regenerated when the model project is built. This means you no longer need to remember to re-run the `dotnet ef dbcontext optimize` command after your EF model changes. See [Auto-compiled models](https://learn.microsoft.com/ef/core/what-is-new/ef-core-9.0/whatsnew#auto-compiled-models) in the _What's New_ docs for more information.

### Sugar for `HierarchyId` path generation

First class support for the SQL Server `HierarchyId` type was [added in EF8](https://learn.microsoft.com/ef/core/providers/sql-server/hierarchyid). In EF9, a sugar method has been added to make it easier to create new child nodes in the tree structure. See [Sugar for HierarchyId path generation](https://learn.microsoft.com/ef/core/what-is-new/ef-core-9.0/whatsnew#hierarchyid-path-generation) in the _What's New_ docs for more information.

### Everything else in preview 3

Preview 3 contains:

- [6 enhancements](https://github.com/dotnet/efcore/issues?q=is%3Aissue+milestone%3A9.0.0-preview3+is%3Aclosed+label%3Atype-enhancement+)
- [5 bug fixes](https://github.com/dotnet/efcore/issues?q=is%3Aissue+milestone%3A9.0.0-preview3+is%3Aclosed+label%3Atype-bug)

EF Core 9 preview 3 (and Microsoft.Data.Sqlite) contains work from the EF Team at Microsoft (@roji @AndriySvyryd @maumar @ajcvickers @cincuranet @SamMonoRT @luisquintanilla) as well as contributions from the EF Core community. The community PRs in EF9 Preview 3 are:

- @lauxjpn: [Cleanup duplicate tests (same test ID)](https://github.com/dotnet/efcore/pull/33185)
- @clement911: [Fixed typo in getting-and-building-the-code.md](https://github.com/dotnet/efcore/pull/33166)
- @Rezakazemi890: [32943-Sugar for HierarchyId path generation](https://github.com/dotnet/efcore/pull/33062)
- @lauxjpn: [Change Where_math(f)_log_new_base queries, so they fail if LOG() parameters are swapped](https://github.com/dotnet/efcore/pull/33342)
- @SteSinger: [33196 command timeout allow zero](https://github.com/dotnet/efcore/pull/33198)

**Comment by ajcvickers** [comment]
@jamesmontemagno Release notes draft in comment above.

File: entity-framework/core/what-is-new/ef-core-9.0/whatsnew.md [hunk 1]
```diff
@@ -562,6 +562,124 @@ protected override void Up(MigrationBuilder migrationBuilder)
...
+
+| MSBuild property   | Description                                                                                                                                                                                                     |
+|--------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| EFOptimizeContext  | Set to `true` to enable auto-compiled models.                                                                                                                                                                   |
+| DbContextName      | The DbContext class to use. Class name only or fully qualified with namespaces. If this option is omitted, EF Core will find the context class. If there are multiple context classes, this option is required. |
+| EFStartupProject   | Relative path to the project folder of the startup project. Default value is the current folder.                                                                                                                |
```
Comment (AndriySvyryd) [inline entity-framework/core/what-is-new/ef-core-9.0/whatsnew.md:638]:
```suggestion
| EFStartupProject   | Relative path to the startup project. Default value is the current folder.                                                                                                                |
```

File: entity-framework/core/what-is-new/ef-core-9.0/whatsnew.md [hunk 2]
```diff
@@ -562,6 +562,124 @@ protected override void Up(MigrationBuilder migrationBuilder)
...
+>> EF is building the model...
+Successfully generated a compiled model, it will be discovered automatically, but you can also call 'options.UseModel(BlogsContextModel.Instance)'. Run this command again when the model is modified.
+PS D:\code\EntityFramework.Docs\samples\core\Miscellaneous\NewInEFCore9.CompiledModels\Model> 
+```
+
+Notice that the log output indicates that the _model was built when running the command_. If we now run the application again, after rebuilding but without making any code changes, then the output is:
+
+```output
+Starting application...
+Model loaded model with 2 entity types.
```
Comment (AndriySvyryd) [inline entity-framework/core/what-is-new/ef-core-9.0/whatsnew.md:608]: We should fix this string

**Review by AndriySvyryd — APPROVED** [review]

File: entity-framework/core/what-is-new/ef-core-9.0/whatsnew.md (hunk 2 above)
Comment (ajcvickers) [reply entity-framework/core/what-is-new/ef-core-9.0/whatsnew.md:608]: @AndriySvyryd Fix in what way?

Comment (ajcvickers) [reply entity-framework/core/what-is-new/ef-core-9.0/whatsnew.md:608]: Oh. Can't see the forest for the trees... :-)

**Pull request merged**